appmgr/src/install/mod.rs

```diff
@@ -192,6 +204,173 @@ pub async fn install(
+                }
+                _ => {
+                    return Err(Error::new(
+                        eyre!("Cannot install over a package in a transient state"),
+                        crate::ErrorKind::InvalidRequest,
+                    ))
+                }
+            }
+            pde.save(&mut tx).await?;
+            tx.commit(None).await?;
+            drop(hdl);
+
+            let res = download_install_s9pk(
+                &new_ctx,
+                &manifest,
+                progress,
+                tokio_util::io::StreamReader::new(req.into_body().map_err(|e| {
+                    std::io::Error::new(
+                        match &e {
+                            e if e.is_connect() => std::io::ErrorKind::ConnectionRefused,
+                            e if e.is_timeout() => std::io::ErrorKind::TimedOut,
+                            _ => std::io::ErrorKind::Other,
+                        },
+                        e,
+                    )
+                })),
+            )
+            .await?;
+            Response::builder()
+                .status(StatusCode::OK)
+                .body(Body::empty())
+                .with_kind(ErrorKind::Network)
+        }
+        .boxed()
+    });
+    let cont = RpcContinuation {
+        created_at: Instant::now(), // TODO
+        handler: handler,
+    };
+    // gc the map
+    let mut guard = ctx.rpc_stream_continuations.lock().await;
+    let gced = std::mem::take(&mut *guard)
+        .into_iter()
+        .filter(|(_, v)| v.created_at.elapsed() < Duration::from_secs(30))
+        .collect::<BTreeMap<RequestGuid, RpcContinuation>>();
+    *guard = gced;
+    drop(guard);
+    // insert the new continuation
+    ctx.rpc_stream_continuations
+        .lock()
+        .await
+        .insert(guid.clone(), cont);
+    Ok(guid)
+}
+
+#[instrument(skip(ctx))]
+async fn cli_install(ctx: CliContext, target: String) -> Result<(), RpcError> {
+    if target.ends_with(".s9pk") {
+        let path = PathBuf::from(target);
+
+        // inspect manifest no verify
+        let manifest = crate::inspect::manifest(path.clone(), true, Some(IoFormat::Json)).await?;
+
+        // rpc call remote sideload
+        tracing::debug!("calling package.sideload");
+        let guid = rpc_toolkit::command_helpers::call_remote(
+            ctx,
+            "package.sideload",
+            serde_json::json!({ "manifest": manifest }),
+            PhantomData::<RequestGuid>,
+        )
+        .await?
+        .result?;
+        tracing::debug!("package.sideload succeeded {:?}", guid);
+
+        // hit continuation api with guid that comes back
+        let file = tokio::fs::File::open(path).await?;
+        let content_length = file.metadata().await?.len();
+        let body = Body::wrap_stream(tokio_util::io::ReaderStream::new(file));
+        let client = reqwest::Client::new();
+        let res = client
+            .post(format!("http://localhost:5960/rest/rpc/{}", guid))
```

Comment:
need a better solution for this. can discuss if needed

Reply:
I can't go through nginx right now due to complaints about body sizes being too large. If you have any ideas for dealing with that we can remove the port reference altogether. Though I remember something about you saying you want it to work remotely.